flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientHandlerTest.java

```diff
@@ -237,7 +254,8 @@ static RemoteInputChannel createRemoteInputChannel(
 			SingleInputGate inputGate,
 			PartitionRequestClient client,
 			int initialBackoff,
-			int maxBackoff) throws Exception {
+			int maxBackoff,
+			MemorySegmentProvider memorySegmentProvider) throws Exception {
 		final ConnectionManager connectionManager = mock(ConnectionManager.class);
 		when(connectionManager.createPartitionRequestClient(any(ConnectionID.class)))
```

Comment:
There are already too many utils here to construct the `RemoteInputChannel`, but we still miss the units of `{SingleInputGate, PartitionRequestClient, MemorySegmentProvider}` which could avoid pass `createRemoteInputChannel(inputGate, client, 0, 0, networkBufferPool)` in `CreditBasedPartitionRequestClientHandlerTest`. 

Maybe we need a separate hotfix for refactoring these parts to make `connectionManager.createPartitionRequestClient` call in separate tests if necessary, then we only need one type to construct `RemoteInputChannel`.